core/src/data/mbtilesDataSource.cpp

```diff
@@ -239,7 +240,8 @@ void MBTilesDataSource::openMBTiles() {
             mode = SQLite::OPEN_READWRITE | SQLite::OPEN_CREATE;
         }
 
-        m_db = std::make_unique<SQLite::Database>(m_path, mode);
+        auto mbTilesPath = _platform->getAssetPath(Url(m_path));
+        m_db = std::make_unique<SQLite::Database>(mbTilesPath, mode);
```

Comment:
@tallytalwar Do you know how sqlite behaves when given an empty string as a database path? If it's unclear it would be worth checking above whether the asset path is empty string and not create the db in that case.
Either way we should at least add a warning log for that error case.